### DISCUSSION TITLE
How to skip Cypress Cloud PR status check if not needed

### DISCUSSION THREAD

**Discussion by cgutierrezpa** [cypress-io/cypress#26071] [CI setup]
Hey everyone,

I have this situation in my team right now. So we are using [GitHub integration for Cypress Cloud](https://docs.cypress.io/guides/cloud/github-integration#Enabling-GitHub-integration-for-a-project), so that we can have this `cypress: <group>` [status check](https://docs.cypress.io/guides/cloud/github-integration#Status-checks) on every PR to get the results of the E2E tests as a comment in the PR and also to prevent PR merges if such check fails (plus other checks that we have). As the project grows, we notice that it is inefficient to run E2E tests on every single PR, because some of them include changes that won't affect the result of those tests, like docs, unit test changes, unrelated CI changes, etc. The problem comes when Cypress adds this `Pending` status check state for every single PR, and only updates it when the CI of the PR starts running the tests with the linked `--group` name. So when the CI skips the E2E jobs because the PR doesn't need to run them, the status check will never get updated and it gets stuck in that Pending state forever, making it unable to work with conditional E2E.

Is there a way to achieve this? Any kind of way to mark the check as `Complete` or `Skipped` so that the PR can get merged?

If there is no way to do that right now, is there a way we can self-implement this `cypress: group` check that Cypress Cloud includes out of the box? In the sense of grouping the results of the parallel tests in 1 single status check, that could be skipped on demand?

One thing that I want to mention is that we cannot rely on the status checks of the jobs themselves (without Cypress Cloud integration) because, as we are running the tests in parallel of 5-6 CI jobs, we can have a complete green pass of the entire test suite even if a job fails in the setup steps prior to running the tests, because the entire E2E suite will simply run all the tests with 1 CI job less. This results in having 1 failed status check, while the others are green and all the E2E tests passed successfully, but still preventing the PR from merge.

Thank you for your time!

**Comment by Blacksmoke16** [comment]
Bump. Having something to handle this would be quite handy when also using the merge queue. The use case being wanting to keep our `cypress:default-group` as a required check, but only run them in the merge queue context. This requires the check to pass for it to be added to the queue in the first place.

I'm currently handling this by triggering the GHA with `cypress/e2e/noop.cy.ts` so that it only runs one test. But having something to avoid needing to install Cypress/run anything at all would be nicer.
